domain/secretbackend/service/service.go

```diff
@@ -612,23 +603,19 @@ func (s *Service) RotateBackendToken(ctx context.Context, backendID string) erro
 // GetRevisionsToDrain looks at the supplied revisions and returns any which should be
 // drained to a different backend for the specified model.
 func (s *Service) GetRevisionsToDrain(ctx context.Context, modelUUID coremodel.UUID, revs []coresecrets.SecretExternalRevision) ([]RevisionInfo, error) {
-	activeBackendDetails, err := s.st.GetModelSecretBackendDetails(ctx, modelUUID)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	jujuBackend, err := s.st.GetSecretBackend(ctx, secretbackend.BackendIdentifier{Name: juju.BackendName})
+	internalBackendUUID, activeBackendUUID, err := s.st.GetInternalAndActiveBackendUUIDs(ctx, modelUUID)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	var result []RevisionInfo
 	for _, r := range revs {
 		if r.ValueRef != nil {
-			if r.ValueRef.BackendID == activeBackendDetails.SecretBackendID {
+			if r.ValueRef.BackendID == activeBackendUUID {
 				continue
 			}
 		} else {
 			// Only internal backend secrets have nil ValueRef.
```

Comment:
So then is this an error?

Reply:
No, we just need to ignore.
Updated the comment to make it clearer.